### PR TITLE
Add location tracking and background tracking to iOS sample.

### PR DESCRIPTION
**Pull request by msmollin**
This will help in debugging #1708

### CODE REVIEW

File: toolchains/iOS.framework.cmake [hunk 1]
```diff
@@ -37,7 +37,7 @@ else()
     endif()
 endif()
 
-set(FRAMEWORKS CoreGraphics CoreFoundation QuartzCore UIKit OpenGLES Security CFNetwork GLKit)
+set(FRAMEWORKS CoreGraphics CoreFoundation CoreLocation QuartzCore UIKit OpenGLES Security CFNetwork GLKit)
```
Comment (msmollin) [inline toolchains/iOS.framework.cmake]: I don't know if this is actually needed because it's only in the sample app?

Comment (karimnaaji) [reply toolchains/iOS.framework.cmake]: Yes this changes the dependency of the framework, you could change it here instead https://github.com/tangrams/tangram-es/blob/master/toolchains/iOS.cmake#L8.

Comment (msmollin) [reply toolchains/iOS.framework.cmake]: Ah ok yeah I changed it there as well. So I can remove this one?

Comment (karimnaaji) [reply toolchains/iOS.framework.cmake]: Yes

File: platforms/ios/demo/src/MapViewController.m [hunk 2]
```diff
@@ -199,11 +211,38 @@ - (void)viewDidLoad
...
+    NSLog(@"Locations came in - %@", locations);
+    TGGeoPoint point = TGGeoPointMake(locations[0].coordinate.longitude, locations[0].coordinate.latitude);
+    [self.locationTrackingMarker pointEased:point seconds:1.0 easeType:TGEaseTypeCubic];
+}
+
+- (void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatus:(CLAuthorizationStatus)status {
+    switch (status) {
+        case kCLAuthorizationStatusAuthorizedAlways:
+            // Enable to turn on background location tracking - be aware we don't shut this off ever, so the sample app will consume battery life if allowed to run in the background.
+            // [self beginBackgroundLocationTracking];
```
Comment (karimnaaji) [inline platforms/ios/demo/src/MapViewController.m]: Not sure if this is expected but if I uncomment this I get the following:

```
2017-10-31 11:28:48.201296-0400 tangram[5255:87931] *** Assertion failure in -[CLLocationManager setAllowsBackgroundLocationUpdates:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/CoreLocationFramework_Sim/CoreLocation-2235.0.2.2/Framework/CoreLocation/CLLocationManager.m:626
2017-10-31 11:28:48.203696-0400 tangram[5255:87931] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Invalid parameter not satisfying: !stayUp || CLClientIsBackgroundable(internal->fClient)'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010631f1cb __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x0000000109b4ef41 objc_exception_throw + 48
	2   CoreFoundation                      0x0000000106324362 +[NSException raise:format:arguments:] + 98
	3   Foundation                          0x00000001095f3089 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 193
	4   CoreLocation                        0x000000010677eb67 CLClientRetrieveData + 28048
	5   tangram                             0x00000001052f9269 -[MapViewController beginBackgroundLocationTracking] + 185
	6   tangram                             0x00000001052f933e -[MapViewController locationManager:didChangeAuthorizationStatus:] + 62
	7   CoreLocation                        0x00000001067892a0 CLClientRetrieveData + 70857
	8   CoreLocation                        0x0000000106788f64 CLClientRetrieveData + 70029
	9   CoreLocation                        0x00000001067731f3 CLClientInvalidate + 1048
	10  CoreFoundation                      0x00000001062c220c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
	11  CoreFoundation                      0x00000001062a6a3b __CFRunLoopDoBlocks + 203
	12  CoreFoundation                      0x00000001062a683b __CFRunLoopRun + 2875
	13  CoreFoundation                      0x00000001062a5a89 CFRunLoopRunSpecific + 409
	14  GraphicsServices                    0x000000010d8d49c6 GSEventRunModal + 62
	15  UIKit                               0x0000000106ad9d30 UIApplicationMain + 159
	16  tangram                             0x00000001052f9523 main + 99
	17  libdyld.dylib                       0x000000010a70ed81 start + 1
)
```

Comment (msmollin) [reply platforms/ios/demo/src/MapViewController.m]: Interesting! I'll take a look.

Comment (msmollin) [reply platforms/ios/demo/src/MapViewController.m]: Ah-ha yeah somehow the `UIBackgroundModes` key got omitted from the Info.plist when I uploaded this. Fixed now.

Comment (karimnaaji) [reply platforms/ios/demo/src/MapViewController.m]: Same happens with latest version.

**Comment by msmollin** [comment]
I tested this with the PR I merged in from @karimnaaji with the fix and can confirm it fixes the issues.

**Pull request merged**